### ISSUE TITLE
independence of classification is currently compromised

### ISSUE THREAD

**Issue by AdrianEvans** [zooniverse/Panoptes-Front-End#2079]
It has been noted that a user can identify a subject and thus find talk discussion related to a subject before they have classified the subject. This open up the possibility of introducing bias.

Since we rely on independence of multiple noisy classifiers for good crowd data it may be an idea to close the loophole.

A user can click on the heart to favourite an image or add it to a collection. with favourites open in a second tab the subject discussion in talk can be accessed just three mouse click away from the point that they are first presented with the image. It may be that in other projects the nature of the tasks are more intuitive that users are less likely to do this but for those like fossilfinder the ambiguity and learning curve may mean it is more of a temptation.

An easy solution is to not have the ability to favourite or add an image to a collection until the summary page at the end of each classification.

**Comment by aliburchard** [comment]
We do definitely want to try and keep classifications independent, though there's probably only so much we can do to keep this from happening. I'd love to try and track how often it actually happens that folks find a way to work around the system, but that might require more intensive user tracking than we're comfortable with.

I'm not a huge fan of restricting the favouriting functionality to the summary page because we have projects that actually skip that page, so I'd like to keep as little functionality tied to that page as possible.

**Comment by VVH** [comment]
This is a good point, but short of removing favorite buttons and Talk functionality, I'm not sure that we're able to get around this. Perhaps @vrooje has further thoughts.

**Comment by alexbfree** [comment]
one option might be that while viewing your favourites collection or any unclassified subject, we suppress all subject IDs and comment threads relating to it, until you have classified the subject.

**Comment by mschwamb** [comment]
@alexbfree people favorite images that they see on Talk, so that's not a viable solution because they then would never be able to see comments etc on it. I know the super users peruse Talk so I don't see that as a viable option

**Comment by mschwamb** [comment]
> 'A user can click on the heart to favourite an image or add it to a collection. with favourites open in a second tab the subject discussion in talk can be accessed just three mouse click away from the point that they are first presented with the image. It may be that in other projects the nature of the tasks are more intuitive that users are less likely to do this but for those like fossilfinder the ambiguity and learning curve may mean it is more of a temptation.'

I would say from my own experience that the changes that people going to do this is pretty rare. It's easy to check if this is an issue, you can find out many images were favorited and if a user is favoriting all the time. That information could be provided or checked by the Zooniverse. I would discourage the practice by telling the core volunteers it impacts your results and why you need their unbiased opinion.  Any time there was an issue like someone creating an account to classify the images a second full time, explaining why that hurt the statistics stopped the issue. 

A simpler solution to help deter this might be to add a random sorting to favorites and classifications so that your newest image doesn't show up at the end of the list. Collections and Favorites don't list the subject id, so you have to click on the image ,but would have to find it. Not sure this is needed.

**Comment by alexbfree** [comment]
@mschwamb Good point, I guess this would hide most talk content. Ideally then we'd need to do it the other way round - each time a user tries to view a subject in Talk, we check if they've ever been presented with it in the classification interface. If they have, but no classification exists, then we suppress the comments. Unfortunately, while this would work at the theoretical level, our current infrastructure doesn't support it in practice. Something like that would be possible once the Nero infrastructure is in place.

**Comment by mschwamb** [comment]
@alexbfree I've seen lots of great things from people seeing Talk conversations on subjects they didn't classify. Especially for the superusers that evolve out of classifying, they start to use the Talk hashtag search to find interesting subjects to model, vet, etc.  I'm not sure how you set this up so it's fine from Talk and not fine if you go through a collection.  These people became super users and were looking at other peoples hashtags and doing their own analysis and then commenting on what they did. I think what you're proposing on doing takes a way a huge amount of the science we get from Talk, so I'd be very wary to try and patch this tiny loophole by stopping people from seeing Talk comments if they haven't seen a subject. 

My bet is this issue happens very infrequently. It would be very easy to check. The science teams don't have access to the favorites data, but someone with access to the full database could check.

**Comment by vrooje** [comment]
I agree with @mschwamb here - I think this is probably very rare and will be most likely to affect people who are already invested enough in the project to go through all the extra trouble, so explaining to them why it's so important that they *don't* do it will probably be adequate. And it can be checked after the fact to some extent (though I'm not sure we could tell the difference between someone who favorites everything while classifying but doesn't look at Talk versus does look at Talk during classification). 

I like the spirit of @alexbfree's suggestions too, that we should just add a small barrier so that people can still favorite subjects while classifying, and those subjects won't show up in the Favorites collection until the classification is complete. But I'd be worried that checking each user's entire classification history before showing Favorites would put an undue burden on the servers for a busy project... and if it's only a very rare problem that's not worth it.

What about a time delay in the appearance of a favorite in the Favorites collection, which applies *only* if the favorite is added from the classification interface? Just a delay of a couple of minutes should be enough for most people to complete their classifications before they can see the new subject, or to deter people who might otherwise find it very easy to check what others have said.

**Comment by alexbfree** [comment]
Meg, re your "seen lots of great things" comment, I think you've misunderstood my second suggestion, "fine from Talk and not fine if you go through a collection" is not what I was saying. I am agreeing with your initial observation that we need to preserve visibility in talk and that therefore my initial proposal wouldn't work. I was saying that it would only be hidden if (a) the user has seen it in the classify interface and (b) not yet classified it, and if those conditions are true it would be hidden everywhere (except classify, of course).
Brooke is right that with the current backend this is too costly to do, though this is exactly the sort of scenario Nero is designed to solve, so it will be easy once we have that.

But I like Brooke's idea of not counting the favourite as visible in your favourites collection or other collection until you have classified it yourself. You're right that this would also require checking the user's classification history, until we have Nero, so is not an option right now. 

The time delay is clever, but i don't like the fact that it doesn't catch everything. I'd rather do a robust solution if we can, then we can truly claim that our classifications are independent in all circumstances.

Maybe the compromise is we do the time delay for now, then something more robust once Nero is in place.

**Comment by mschwamb** [comment]
@alexbfree  at least in the old versions of Talk, so that's why I'm using that terminology. Collections were a feature of Talk so that's why I reference it that way. I understand that you're suggesting to strip away the ability to go from a collection click on the image and see all the discussions and hashtags associated with that subject on the subjects's talk page that are shown on the subject page unless you've classified the image (and that this only applies if you get there through collections)

 Some of the Talk super users  on my previous projects used collections to curate their best candidates so not being able to go to see Talk  discussions/relevant info on the Talk subject pages going through collections I think is making Talk less effective and makes collections much less useful. These super users would also go through other people's collections and identify interesting things. They would look at the Talk comments to see if one of of the other super users had vetted it and identified the light curve as a false positive.

**Comment by alexbfree** [comment]
I think i'm confused then. Are you talking about a deficiency of Ouroboros or Panoptes? Don't both allow you to view others collections and add them to your own and curate your candidates? Or are you talking about the lack of commenting on collections in Panoptes?

**Comment by vrooje** [comment]
> I'd rather do a robust solution if we can, then we can truly claim that our classifications are independent in all circumstances.

I'd always rather a robust solution, but the cost/benefit here doesn't really work out for me unless I'm massively misunderstanding how much effort this would take and what strain it might put on the system (especially in times of high traffic). 

Besides, we can never really claim our classifications are independent in all circumstances. Our volunteers are incredibly resourceful and have in the past found ways we hadn't thought of to get additional information about subjects while classifying; I have no doubt it will happen again. And even if it doesn't, we can't prevent random occurrences, like someone noticing the subject they were just served was also the subject they just saw a discussion about on Talk. We can't promise a 100% solution. What we can (and do) do is make all reasonable efforts to minimize this kind of contamination.

**Comment by alexbfree** [comment]
> What we can (and do) do is make all reasonable efforts to minimize this kind of contamination.

Agreed. Re cost/benefit, my point is that equation is very different once we have Nero, the effort then will be much less and there wouldn't be strain on the system as it maintains a record of what has been done. Also apparently even without Nero, it's easy to query whether a particular subject has been classified by a particular user without looking in their entire history. In short, we should be able to largely close this loophole one way or another, without a massive amount of effort, definitely in the long term and maybe in the short term too.

**Comment by camallen** [comment]
@alexbfree can you outline some api end points you'd use to block a user accessing content they shouldn't?

**Comment by alexbfree** [comment]
@camallen are you asking me about whether we should block access to data at the API level and which ones we should block, or are you asking me which existing endpoints would allow us to query user's classification to decide when to hide things?

**Comment by camallen** [comment]
the later, i'm trying to figure out how you're going to use the api / db state to stop people working around the system with different logins, read-only access, etc.

**Comment by alexbfree** [comment]
@camallen first i'm not talking about stopping a determined circumventer, i'm talking about stopping casual/accidental contamination of viewpoint in the UI layer. Secondly I don't know the specifics but @marten mentioned that classifications are stored with various IDs associated so it should be possible to look this up more easily. (It's possible I misunderstood him).

**Comment by camallen** [comment]
> my point is that equation is very different once we have Nero,

Ok - i'm not sure what nero is going to do in this context. Might make it easier to understand if we have an idea of the sort of state you'd need to make the UI changes. If you have all the required information pieces then you can modify the UI state accordingly.

**Comment by alexbfree** [comment]
the only things we need to know to achieve my solution are (a) has this user been served this image? (could potentially track in browser or could do via backend) and (b) has this user classified this image?
My understanding is that both become easy in Nero+Panoptes-land. Again, I may be wrong!

**Comment by chrislintott** [comment]
I think this is unnecessary load on the API for a problem which, while important in theory, makes very little difference in practice.

**Comment by camallen** [comment]
>(a) has this user been served this image?

Already provided via the api response (similar to retired / finished_at state). Note this is per-workflow only so I may have seen an image before e.g. for a collaborative filtering task.

> (b) has this user classified this image?

Tied to a), we only record that a user has submitted a classification for an image not that they have been sent an image for classification.

**Comment by marten** [comment]
> Ok - i'm not sure what nero is going to do in this context.

Neither am I. Seems out of scope to me.

**Comment by MvGulik** [comment]
As a person cursed with a certain level of perfection. I'm thinking that "unless there is actual data that's showing that the mentioned contamination actually results in a measurable bias in the collected result data." There is a change your trying to fix a perceived general issue, ... that might actually not exist or matter.
Just a thought.

**Comment by AvastMH** [comment]
I think that MvGulik has expressed the central element that I was going to raise.

Quick question - what is Nero and what will it do for us please?

 :confused:

**Comment by jelliotartz** [comment]
Closing as this has not received comments or action for over a year.
